fix(kong_client): return error in fetching listeners when no ready kong clients (#7293)
Kong/kubernetes-ingress-controller: CHANGELOG.md

  where KIC is permitted to access only the given workspace but cannot access
  `/status` endpoint.
  [#7233](https://github.com/Kong/kubernetes-ingress-controller/pull/7233)

> - Return error when no ready Kong admin API clients in fetching listeners in
>   the reconciliation of `Gateway`s to trigger a requeue when no Kong gateways
>   are available.

Reviewer: I think we can reword this to make it a little more clear, how about:

```suggestion
- Report an error and requeue when there are no ready Kong admin API clients
  during the reconciliation of `Gateway`s.
```
